src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+}
+
+void LocomotorStore::Parse_Locomotor_Template_Definition(INI *ini)
+{
+    if (g_theLocomotorStore == nullptr) {
+        throw CODE_06;
+    }
+
+    bool found = false;
+    const char *name = ini->Get_Next_Token();
+    NameKeyType key = g_theNameKeyGenerator->Name_To_Key(name);
+
+    LocomotorTemplate *t = g_theLocomotorStore->Find_Locomotor_Template(key);
+    if (t != nullptr) {
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            t = g_theLocomotorStore->New_Override(static_cast<LocomotorTemplate *>(t->Friend_Get_Final_Override()));
+        }
+
+        found = true;
+    } else {
+        t = NEW_POOL_OBJ(LocomotorTemplate);
+
+        if (ini->Get_Load_Type() == INILoadType::INI_LOAD_CREATE_OVERRIDES) {
+            t->Set_Is_Allocated();
+        }
+    }
+
+    t->Set_Name(name);
+    ini->Init_From_INI(t, LocomotorTemplate::Get_Field_Parse());
+    t->Validate();
+
+    if (!found) {
+        g_theLocomotorStore->m_locomotorTemplates[key] = t;
+    }
+}
+
+Locomotor::Locomotor(const LocomotorTemplate *tmpl) :
+    m_brakingFactor(1.0f),
+    m_maxLift(99999.0f),
+    m_maxSpeed(99999.0f),
+    m_maxAccel(99999.0f),
+    m_maxBreaking(99999.0f),
+    m_maxTurnRate(99999.0f),
+    m_flags(0)
+{
+    m_template = tmpl;
+    m_closeEnoughDist = m_template->m_closeEnoughDist;
+    Set_Flag(CLOSE_ENOUGH_DIST_3D, m_template->m_closeEnoughDist3D);
+    m_preferredHeight = m_template->m_preferredHeight;
+    m_preferredHeightDamping = m_template->m_preferredHeightDamping;
+    m_wanderAngle = Get_Logic_Random_Value_Real(DEG_TO_RADF(-30), DEG_TO_RADF(30));
+    m_wanderLength = Get_Logic_Random_Value_Real(0.80000001f, 1.2f);
+    Set_Flag(WANDER_DIRECTION, (Get_Logic_Random_Value(0, 1) != 0));
+    m_moveFrame = g_theGameLogic->Get_Frame() + 2.5f * 30.0f;
+}
+
+Locomotor::Locomotor(const Locomotor &that) :
+    m_brakingFactor(that.m_brakingFactor),
+    m_maxLift(that.m_maxLift),
+    m_maxSpeed(that.m_maxSpeed),
+    m_maxAccel(that.m_maxAccel),
+    m_maxBreaking(that.m_maxBreaking),
+    m_maxTurnRate(that.m_maxTurnRate),
+    m_flags(that.m_flags),
+    m_closeEnoughDist(that.m_closeEnoughDist),
+    m_preferredHeight(that.m_preferredHeight),
+    m_preferredHeightDamping(that.m_preferredHeightDamping),
+    m_wanderAngle(that.m_wanderAngle),
+    m_wanderLength(that.m_wanderLength)
+{
+    m_maintainPos.Zero();
+}
+
+Locomotor &Locomotor::operator=(const Locomotor &that)
+{
+    if (this != &that) {
+        m_template = that.m_template;
+        m_brakingFactor = that.m_brakingFactor;
+        m_maxLift = that.m_maxLift;
+        m_maxSpeed = that.m_maxSpeed;
+        m_maxAccel = that.m_maxAccel;
+        m_maxBreaking = that.m_maxBreaking;
+        m_maxTurnRate = that.m_maxTurnRate;
+        m_flags = that.m_flags;
+        m_closeEnoughDist = that.m_closeEnoughDist;
+        m_preferredHeight = that.m_preferredHeight;
+        m_preferredHeightDamping = that.m_preferredHeightDamping;
```

Comment:
`m_maintainPos`, `m_wanderAngle`, `m_wanderLength`, `m_moveFrame` appear to be missing here.

Reply:
These are explicitly not being coped here.